src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/ComponentAccessibilityCodeActionProvider.cs

```diff
@@ -0,0 +1,219 @@
+            {
+                return;
+            }
+
+            var actionParams = new CreateComponentCodeActionParams
+            {
+                Uri = context.Request.TextDocument.Uri,
+                Path = newComponentPath,
+            };
+            var data = JObject.FromObject(actionParams);
+
+            var resolutionParams = new RazorCodeActionResolutionParams
+            {
+                Action = LanguageServerConstants.CodeActions.CreateComponentFromTag,
+                Data = data,
+            };
+            var serializedParams = JToken.FromObject(resolutionParams);
+            var arguments = new JArray(serializedParams);
+
+            container.Add(new CommandOrCodeAction(new Command
+            {
+                Title = "Create component from tag",
+                Name = "razor/runCodeAction",
+                Arguments = arguments,
+            }));
+        }
+
+        private void AddComponentAccessFromTag(RazorCodeActionContext context, MarkupStartTagSyntax startTag, List<CommandOrCodeAction> container)
+        {
+            // Get all data necessary for matching
+            var tagName = startTag.Name.Content;
+            string parentTagName = null;
+            if (startTag.Parent.Parent is MarkupElementSyntax parentElement)
+            {
+                parentTagName = parentElement.StartTag.Name.Content;
+            }
+            var attributes = _tagHelperFactsService.StringifyAttributes(startTag.Attributes);
+
+            // Find all matching tag helpers
+            var matching = new Dictionary<string, TagHelperPair>();
+            foreach (var tagHelper in context.DocumentSnapshot.Project.TagHelpers)
+            {
+                if (tagHelper.TagMatchingRules.All(rule => TagHelperMatchingConventions.SatisfiesRule(tagName, parentTagName, attributes, rule)))
+                {
+                    matching.Add(tagHelper.Name, new TagHelperPair { Short = tagHelper });
+                }
+            }
+
+            // Iterate and find the fully qualified version
+            foreach (var tagHelper in context.DocumentSnapshot.Project.TagHelpers)
+            {
+                if (matching.ContainsKey(tagHelper.Name))
+                {
+                    var tagHelperPair = matching[tagHelper.Name];
+                    if (tagHelperPair != null && tagHelper != tagHelperPair.Short)
+                    {
+                        tagHelperPair.FullyQualified = tagHelper;
+                    }
+                }
+            }
+
+            // For all the matches, add options for add @using and fully qualify
+            foreach (var tagHelperPair in matching.Values)
+            {
+                DefaultRazorTagHelperBinderPhase.ComponentDirectiveVisitor.TrySplitNamespaceAndType(tagHelperPair.Short.Name, out var namespaceSpan, out var typeSpan);
+                var namespaceName = tagHelperPair.Short.Name.Substring(namespaceSpan.Start, namespaceSpan.Length);
```

Comment:
I believe there's a library function (with `TrySplitNamespaceAndType`) which handles this for us.